common/src/main/java/com/sequenceiq/cloudbreak/quartz/configuration/TransactionalScheduler.java

```diff
@@ -0,0 +1,77 @@
+package com.sequenceiq.cloudbreak.quartz.configuration;
+
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.ListenerManager;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.impl.matchers.GroupMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.service.TransactionService;
+import com.sequenceiq.cloudbreak.common.service.TransactionService.TransactionExecutionException;
+
+@Component
+public class TransactionalScheduler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransactionalScheduler.class);
+
+    @Inject
+    private Scheduler scheduler;
+
+    @Inject
+    private TransactionService transactionService;
+
+    public void clear() throws TransactionExecutionException {
+        transactionService.required(() -> {
+            try {
+                scheduler.clear();
+            } catch (SchedulerException e) {
+                LOGGER.error("Scheduler clear failed.", e);
+                throw new RuntimeException(e);
```

Comment:
It could have its own exception type (as Scheduler has ScheduledException)
Same for the below methods.